include/SFML/Network/IpAddress.hpp

```diff
@@ -43,47 +44,40 @@ namespace sf
 ////////////////////////////////////////////////////////////
 class SFML_NETWORK_API IpAddress
 {
-public:
+private:
+    struct NoByteOrderConversionTag { };
 
     ////////////////////////////////////////////////////////////
-    /// \brief Default constructor
-    ///
-    /// This constructor creates an empty (invalid) address
+    /// \brief Construct the address from a 32-bits integer
     ///
-    ////////////////////////////////////////////////////////////
-    IpAddress();
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Construct the address from a string
+    /// This constructor directly sets the internal representation
+    /// of the IP address without performing any byte order conversion.
     ///
-    /// Here \a address can be either a decimal address
-    /// (ex: "192.168.1.56") or a network name (ex: "localhost").
+    /// \param address 4 bytes of the address packed into a 32-bits integer
     ///
-    /// \param address IP address or network name
+    /// \see toInteger
     ///
     ////////////////////////////////////////////////////////////
-    IpAddress(const std::string& address);
+    explicit IpAddress(NoByteOrderConversionTag, Uint32 address);
 
+public:
     ////////////////////////////////////////////////////////////
-    /// \brief Construct the address from a string
+    /// \brief Construct the address from a null-terminated string view
     ///
     /// Here \a address can be either a decimal address
     /// (ex: "192.168.1.56") or a network name (ex: "localhost").
-    /// This is equivalent to the constructor taking a std::string
-    /// parameter, it is defined for convenience so that the
-    /// implicit conversions from literal strings to IpAddress work.
     ///
     /// \param address IP address or network name
     ///
     ////////////////////////////////////////////////////////////
-    IpAddress(const char* address);
+    static std::optional<IpAddress> create(std::string_view address);
```

Comment:
If we use the [Named Constructor Idiom](https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Named_Constructor), should we not give it a different name, like `fromString()`?

Reply:
I am up for suggestions. `resolve` was also another idea, but `fromString` sounds good to me. @eXpl0it3r, @ChrisThrasher, thoughts?

Reply:
`resolve` is my favorite name so far.

Reply:
"resolve" sounds a bit like an actual connection to the IP address is established.

Reply:
I don't know... if one isn't familiar with how names are _resolved_ to addresses during name resolution, maybe that is something to read into before starting with network programming. 😉

Reply:
Also, calling it `fromString` is misleading. The almost exclusively used dotted decimal notation is the de facto standard way of representing an IPv4 address. The user might think that `fromString` can only be used to convert an IPv4 address in its textual representation to an `sf::IpAddress`. The function in question performs a complete name resolution.

Reply:
Will change the name to `resolve`.